packages/mobile/src/import/saga.ts

```diff
@@ -32,41 +45,71 @@ import { assignAccountFromPrivateKey, waitWeb3LastBlock } from 'src/web3/saga'
 
 const TAG = 'import/saga'
 
+export const MAX_BALANCE_CHECK_TASKS = 5
+export const MNEMONIC_AUTOCORRECT_TIMEOUT = 5000 // ms
```

Comment:
Instead of gating this to X seconds, have you considered trying the Y most likely phrases? I'm worried that this may not work well under poor network conditions.
Maybe this worry is unwarranted. To check this something we could do is send an analytics event with the number of suggestions tried and see if this returns 0 or a very low number with some frequency.

Reply:
My choice to use on the timeout is because we don't want to have the user wait too long before we report what words are in error so they can fix them manually. If there was a minimum number of attempts, but no timeout (or a very high timeout) then the feedback would take too long when internet connections are slow. If we have a maximum number of attempts, then we are not doing the most we can to correct a phrase. Another way to say this is I think the timeout is most closely aligned with the UX cost we are trying to minimize, and so should be the metric we use to cut our losses and give up on correction.

I do agree that analytics can help address these assumptions. I've added some events that should help answer the questions of how to make this feature work best.